internals/secrethub/tree.go

```diff
@@ -47,42 +50,69 @@ func (cmd *TreeCommand) Register(r command.Registerer) {
 	clause := r.Command("tree", "List contents of a directory in a tree-like format.")
 	clause.Arg("dir-path", "The path to to show contents for").Required().PlaceHolder(optionalDirPathPlaceHolder).SetValue(&cmd.path)
 
+	clause.Flag("full-paths", "Print the full paths of the directories and secrets.").Short('f').BoolVar(&cmd.fullPaths)
+	clause.Flag("no-indentation", "Print the content without indentation.").Short('i').BoolVar(&cmd.noIndentation)
+	clause.Flag("no-report", "Skip the report at the bottom.").BoolVar(&cmd.noReport)
+
 	command.BindAction(clause, cmd.Run)
 }
 
 // printTree recursively prints the tree's contents in a tree-like structure.
-func printTree(t *api.Tree, w io.Writer) {
+func (cmd *TreeCommand) printTree(t *api.Tree, w io.Writer) {
 	name := colorizeByStatus(t.RootDir.Status, t.RootDir.Name)
-	fmt.Fprintf(w, "%s/\n", name)
-
-	printDirContentsRecursively(t.RootDir, "", w)
-
-	fmt.Fprintf(w,
-		"\n%s, %s\n",
-		pluralize("directory", "directories", t.DirCount()),
-		pluralize("secret", "secrets", t.SecretCount()),
-	)
+	fmt.Fprintf(w, "%s\n", name)
+
+	var format [4]string
+	if cmd.noIndentation {
+		format[0] = "%s%s%s\n"
+		format[1] = "%s%s%s\n"
+		format[2] = ""
+		format[3] = ""
+	} else {
+		format[0] = "%s└── %s%s\n"
+		format[1] = "%s├── %s%s\n"
+		format[2] = "    "
+		format[3] = "│   "
+	}
+	if cmd.fullPaths {
+		cmd.printDirContentsRecursively(t.RootDir, "", w, format, t.RootDir.Name)
```

Comment:
To print the full path, I believe we need to pass the full path of the root directory, instead of the name:

```suggestion
		cmd.printDirContentsRecursively(t.RootDir, "", w, format, cmd.path.Value())
```

Currently, when I do:

```
secrethub tree simon/test
```

I'm seeing this tree:
```
test
├── test/directory
│   └── test/directory/secret
└── test/s1

1 directory, 2 secrets
```

And I would expect this tree:
```
simon/test
├── simon/test/directory
│   └── simon/test/directory/secret
└── simon/test/s1

1 directory, 2 secrets
```

Reply:
This is actually something that I and @edif2008 were discussing today, about how to grab the namespace of the repo! Thanks for the tip!

Reply:
This should now be fixed.